### PR TITLE
chore: [testing] Mic/todos in moderators fix

### PR DESCRIPTION
**Pull request by michaelneale**
adding in fixes from https://github.com/square/exchange/pull/54 to try to fix broken test on https://github.com/square/exchange/pull/52

Also, we can't default to contextsummarizer while it hard codes gpt4o https://github.com/square/exchange/issues/57

### CODE REVIEW

**Review by codefromthecrypt — COMMENTED** [review]
🚢

**Comment by codefromthecrypt** [comment]
actually this has the wrong model thing again..

```
>               raise httpx.HTTPStatusError(f"{e}\n{response.text}", request=e.request, response=e.response)
E               httpx.HTTPStatusError: Client error '404 Not Found' for url 'http://localhost:11434/v1/chat/completions'
E               For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404
E               {"error":{"message":"model \"gpt-4o-mini\" not found, try pulling it first","type":"api_error","param":null,"code":null}}
```

**Comment by michaelneale** [comment]
@codefromthecrypt hrm - yes same as last time. Is this same bug? I am totally confused

File: src/exchange/exchange.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ class Exchange:
     provider: Provider
     model: str
     system: str
-    moderator: Moderator = field(default=ContextTruncate())
+    moderator: Moderator = field(default=ContextSummarizer())
```
Comment (codefromthecrypt) [inline src/exchange/exchange.py:43]: let's remove this default so there's no chance of setting it wrong (e.g. better to error due to not inheriting model than using wrong one)

```
class ContextSummarizer(ContextTruncate):
    def __init__(
        self,
        model: str = "gpt-4o-mini",
```

Comment (michaelneale) [reply src/exchange/exchange.py:43]: yes, or it needs to be fixed to take in the correct model

**Pull request closed without merging**